### PR TITLE
Adds topic to show output document examples for Topbeat

### PR DESCRIPTION
**Pull request by dedemorton**
@monicasarbu @ruflin Check out this topic and let me know what you think (for issue #484 ). Not sure about the title of the topic, but you can see how we might show the structure without necessarily making promises that we'll adhere to the structure. WDYT?

### CODE REVIEW

File: topbeat/docs/json.asciidoc [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+----------------------------------------------------------------------
+{
+  "@timestamp": "2015-12-09T22:47:13.663Z",
+  "beat": {
+    "hostname": "MacBook-Pro.local",
+    "name": "MacBook-Pro.local"
+  },
+  "count": 1,
+  "fs": {
```
Comment (ruflin) [inline topbeat/docs/json.asciidoc]: It would be nice here to have an example with at least 2 FS to make it obvious it is a list.

**Comment by ruflin** [comment]
LGTM. @monicasarbu what do you think?

**Comment by tsg** [comment]
To be honest I don't know if maintaining these in the docs is worth the effort. What would the users need this for and why couldn't they just use kibana to view them. I'd rather invest in adding topbeat to our demo, then we can point people to that if they want to see example JSON documents.

**Comment by ruflin** [comment]
@tsg As mentioned before here, I think it is really useful (at least to me): https://github.com/elastic/beats/issues/484#issuecomment-163396109 Having to go to Kibana and know how to use Kibana to just get this information seems to be overkill to me.

**Comment by monicasarbu** [comment]
I suggest to create for each Beat another file in GitHub with an example of the json object and "link" to it in the README file or from the docs as an example output. I don't like the idea of having two different sections in the docs describing the same thing: Json example and Exported fields as I consider the json object an example of the exported fields.

**Comment by monicasarbu** [comment]
This json file might be tricky to generate to be similar with the "real output" because values are changing (integer to float), fields are removed or added. The best will be to copy and paste it from Kibana. This is what I did for Topbeat.

**Comment by ruflin** [comment]
@monicasarbu Sounds like a good idea. The JSON is definitive a specific example instead of the abstract description. Example data often say much more then just some descriptions. What I'm not sure how we will create a good example file out of the fields.yml file, for example with multiple CPU's etc.

**Comment by ruflin** [comment]
@monicasarbu +1. Alternative copy one from a system test output.

**Comment by tsg** [comment]
IMHO, best would be to use the integration tests generate one. But then you have to commit the results.

**Comment by monicasarbu** [comment]
+1 for inetegration tests

**Comment by dedemorton** [comment]
+1 for JSON docs generated from system tests

I don't agree that providing an example of the exported fields is duplicating what we cover in the exported fields topic (though admittedly the JSON examples belong after the topic on exported fields). Sometimes people like to scan documentation to get a high level view of what a thing does (especially when the doc is short). Having examples makes that a little easier. 

But I'm OK with the idea of linking to examples in GitHub, especially if we can generate examples that are always current.

**Comment by monicasarbu** [comment]
Let's start small with creating first the file in the GitHub repo and then try to automate it and if everything works fine, we could also consider adding it to the docs.

**Comment by ruflin** [comment]
@dedemorton is there a follow up issue to this?

**Comment by dedemorton** [comment]
@ruflin I've added a comment to https://github.com/elastic/beats/issues/484 so we can track this work in the original issue. I closed this PR because I  thought we wouldn't want to merge it.

**Comment by ruflin** [comment]
@dedemorton @monicasarbu Not sure if I understand this fully. As long as we don't have it in the docs, I don't see any benefit of having the json file at all. So I would suggest to go the other way around: First have it in the docs (manually generated) and the figure out how to automate it if it gets too much of a pain. Don't get me wrong, I'm all for automating, but for me it looks like it will take as first 5 minutes to do it manually and 2h+ to get it automated. My suggestion is to go with the simple solution first and directly have the benefit.

**Comment by dedemorton** [comment]
Re-opening this PR because it sounds like we need more discussion.

**Comment by ruflin** [comment]
I would like to move forward here and would suggest to merge this PR so we have a first version in the docs and can iterate from there. Thoughts?

**Comment by dedemorton** [comment]
@monicasarbu What you do think about Ruflin's proposal to merge this?

**Comment by monicasarbu** [comment]
@tsg what do you think about adding a json object example to the docs?

**Comment by tsg** [comment]
@monicasarbu Not a fan either, but I agree to try it and see how much of a burden keeping it up to date becomes.

**Comment by ruflin** [comment]
As this is already quite old, I worry it will break our docs when I merge it. I suggest to reopen this PR against master so we have this in future release to test (v2.0). @dedemorton Could you rebase it on top of master and open a PR against master?

**Comment by urso** [comment]
+1

**Comment by dedemorton** [comment]
@ruflin Closing this PR (there's a related issue still open to track the work). I'll submit a new PR later this week.

**Comment by ruflin** [comment]
@dedemorton :+1:

**Pull request closed without merging**